domain.c

```diff
@@ -215,12 +243,46 @@ void domain_maintain(void)
     force_tree_rebuild();
 }
 
+/* this function generates several domain decomposition policies for attempting
+ * creating the domain. */
+static int
+domain_policies_init(DomainDecompositionPolicy policies[],
+        const int NincreaseAlloc,
+        const int NdecreaseAccuracy,
+        const int SwitchToGlobal)
+{
+    int i;
+    for(i = 0; i < NincreaseAlloc; i ++) {
+        policies[i].TopNodeAllocFactor = All.TopNodeAllocFactor * pow(1.3, i);
+        policies[i].UseGlobalSort = All.DomainUseGlobalSorting;
+        policies[i].PreSort = 0;
+        policies[i].SubSampleDistance = 16;
+    }
+
+    for(i = NincreaseAlloc; i < NincreaseAlloc + NdecreaseAccuracy; i ++) {
+        policies[i].TopNodeAllocFactor = policies[NincreaseAlloc].TopNodeAllocFactor;
+        policies[i].UseGlobalSort = All.DomainUseGlobalSorting;
+        policies[i].PreSort = 0;
+        policies[i].SubSampleDistance = 16 * (i - NincreaseAlloc);
```

Comment:
I don't understand this line - why would increasing the sampling distance make a formerly failing domain decomposition succeed?

Reply:
I don't think it will do, though I thought so. I'll change this back to just increasing TopNodeAllocFactor for now.